src/Qir/Runtime/stdlib/src/strings.rs

```diff
@@ -0,0 +1,407 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#![deny(clippy::all, clippy::pedantic)]
+
+use crate::update_counts;
+use crate::Pauli;
+use num_bigint::BigInt;
+use std::{
+    ffi::{CStr, CString},
+    os::raw::{c_char, c_double},
+    rc::Rc,
+};
+
+/// # Safety
+///
+/// This function should only be called with a valid, null-terminated, C-style string.
```

Comment:
Feels very funny somehow that, even in quantum, we're still carrying around C-style strings. Ah, well.

Reply:
Agreed... the legacy that cannot be escaped!